code/game/machinery/vendors/vending.dm

```diff
@@ -312,7 +331,19 @@
 	else
 		..()
 
+/obj/machinery/economy/vending/AltClick(mob/user)
+	if(!tilted || !Adjacent(user) || user.incapacitated())
```

Comment:
Check for handsblocked not incapacitated. If you don't want them to do it while laying down check for body position aswell